client/app/controllers/collection/add.js

```diff
@@ -1,19 +1,55 @@
 import Ember from 'ember';
 
+// Helper function to determine if discipline has changed (comparing list of lists)
+function disciplineArraysEqual(a, b) {
+    if (a === b) return true;
+    if (a == null || b == null) return false;
+    if (a.length !== b.length) return false;
+
+    for (let i = 0; i < a.length; ++i) {
+        if (a[i].length !== b[i].length) return false;
+        for (let j = 0; j < a[i].length; ++j) {
+            if (a[i][j] !== b[i][j]) return false;
+        }
+    }
+    return true;
+}
+
+function subjectIdMap(subjectArray) {
+    // Maps array of arrays of disciplines into array of arrays of discipline ids.
+    return subjectArray.map(subjectBlock => subjectBlock.map(subject => subject.id));
+}
+
 export default Ember.Controller.extend({
     addMethod: 'select', // 'select' or 'create'
     methodSelected: false,
     type: Ember.computed('model.settings', function() {
         var collectionType = this.get('model.settings.collectionType') || 'project';
         return collectionType.toLowerCase();
     }),
+    editMode: false,
+    disciplineModifiedToggle: false,
+    disciplineChanged: true,
+    subjectsList: ['Math', 'Science'],
```

Comment:
I don't think this is the correct format for subjects since I'm getting an error in `subjectIdMap` so I'll look into what it is in preprints